### ISSUE TITLE
MvcMatchers: UnsupportedOperationException: Section 4.4 of the Servlet 3.0 specification does not permit this method to be called from a ServletContextListener that was not defined in web.xml, a web-fragment.xml file nor annotated with @WebListener

### ISSUE THREAD

**Issue by bjornharvold** [spring-projects/spring-security#4027]
### Summary
Upgraded to v4.1.2 of Spring Security. Application no longer starts siting an issue with the mvcMatchers.

### Actual Behavior
This code has worked great since I started using it but fails in 4.1.2.

### Expected Behavior
Start up my Spring web app and have Spring Security secure my urls using mvcMatchers

### Configuration
```
private static final String API_ADMINISTRATION_URL = "/api/administration/**";
private static final String API_MANAGER_URL = "/api/manager/**";
private static final String API_USER_URL = "/api/secured/**";
private static final String API_AUTHENTICATED_URL = "/api/authenticated/**";

@Override
    protected void configure(HttpSecurity http) throws Exception {
        log.info("Configuring springSecurityFilterChain...");

        // header details
        http
                .cors()
                .and()
                .headers()
                .frameOptions()
                .sameOrigin()
                .cacheControl()
                .and().xssProtection()
                .and().contentTypeOptions()
                .and().httpStrictTransportSecurity()
        ;

        http
                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);

        // the url patterns to secure
        http
                .authorizeRequests()
                .mvcMatchers(API_AUTHENTICATED_URL).hasAnyRole("ADMIN", "MANAGER", "USER")
                .mvcMatchers(API_ADMINISTRATION_URL).hasRole("ADMIN")
                .mvcMatchers(API_MANAGER_URL).hasRole("MANAGER")
                .mvcMatchers(API_USER_URL).hasRole("USER")
                .mvcMatchers(API_LOGIN_URL).anonymous()
                .mvcMatchers(API_PUBLIC_URL).permitAll()
        ;

        // new logout handler from spring 4.0.2 - maybe nice to have. Has not been tested yet.
//        HttpStatusReturningLogoutSuccessHandler logoutHandler = new HttpStatusReturningLogoutSuccessHandler();
//        http
//                .logout().logoutSuccessHandler(logoutHandler);

        // filter details
        http
                .addFilterAfter(authFilter(), ApiRequestHeaderAuthenticationFilter.class)
                .addFilter(preAuthFilter());
        http
                .csrf().disable();

        http
                .exceptionHandling().authenticationEntryPoint(new UnauthorizedEntryPoint(objectMapper));
    }
```

### Version

v4.1.2. Works fine on all earlier versions where mvcMatchers is a feature.

### Stacktrace
```
2016-08-17 09:46:55,864 E ContextLoader - Context initialization failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'springSecurityFilterChain' defined in class path resource [org/springframework/security/config/annotation/web/configuration/WebSecurityConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [javax.servlet.Filter]: Factory method 'springSecurityFilterChain' threw exception; nested exception is java.lang.RuntimeException: Could not postProcess org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher@725dfe92 of type class org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1123) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1018) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:510) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:296) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:776) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:861) ~[spring-context-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:541) ~[spring-context-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:444) ~[spring-web-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:326) [spring-web-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:107) [spring-web-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4714) [catalina.jar:8.5.4]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5178) [catalina.jar:8.5.4]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:152) [catalina.jar:8.5.4]
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:724) [catalina.jar:8.5.4]
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:700) [catalina.jar:8.5.4]
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:734) [catalina.jar:8.5.4]
	at org.apache.catalina.startup.HostConfig.manageApp(HostConfig.java:1702) [catalina.jar:8.5.4]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0]
	at org.apache.tomcat.util.modeler.BaseModelMBean.invoke(BaseModelMBean.java:300) [tomcat-coyote.jar:8.5.4]
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819) [?:1.8.0]
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801) [?:1.8.0]
	at org.apache.catalina.mbeans.MBeanFactory.createStandardContext(MBeanFactory.java:482) [catalina.jar:8.5.4]
	at org.apache.catalina.mbeans.MBeanFactory.createStandardContext(MBeanFactory.java:431) [catalina.jar:8.5.4]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0]
	at org.apache.tomcat.util.modeler.BaseModelMBean.invoke(BaseModelMBean.java:300) [tomcat-coyote.jar:8.5.4]
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819) [?:1.8.0]
	at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801) [?:1.8.0]
	at javax.management.remote.rmi.RMIConnectionImpl.doOperation(RMIConnectionImpl.java:1466) [?:1.8.0]
	at javax.management.remote.rmi.RMIConnectionImpl.access$300(RMIConnectionImpl.java:76) [?:1.8.0]
	at javax.management.remote.rmi.RMIConnectionImpl$PrivilegedOperation.run(RMIConnectionImpl.java:1307) [?:1.8.0]
	at javax.management.remote.rmi.RMIConnectionImpl.doPrivilegedOperation(RMIConnectionImpl.java:1399) [?:1.8.0]
	at javax.management.remote.rmi.RMIConnectionImpl.invoke(RMIConnectionImpl.java:828) [?:1.8.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0]
	at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:323) [?:1.8.0]
	at sun.rmi.transport.Transport$1.run(Transport.java:178) [?:1.8.0]
	at sun.rmi.transport.Transport$1.run(Transport.java:175) [?:1.8.0]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0]
	at sun.rmi.transport.Transport.serviceCall(Transport.java:174) [?:1.8.0]
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:557) [?:1.8.0]
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:812) [?:1.8.0]
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:671) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0]
	at java.lang.Thread.run(Thread.java:744) [?:1.8.0]
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [javax.servlet.Filter]: Factory method 'springSecurityFilterChain' threw exception; nested exception is java.lang.RuntimeException: Could not postProcess org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher@725dfe92 of type class org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 59 more
Caused by: java.lang.RuntimeException: Could not postProcess org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher@725dfe92 of type class org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher
	at org.springframework.security.config.annotation.configuration.AutowireBeanFactoryObjectPostProcessor.postProcess(AutowireBeanFactoryObjectPostProcessor.java:72) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry.createMvcMatchers(AbstractRequestMatcherRegistry.java:170) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.mvcMatchers(ExpressionUrlAuthorizationConfigurer.java:118) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.mvcMatchers(ExpressionUrlAuthorizationConfigurer.java:123) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at com.traveliko.platform.server.common.config.WebSecurityConfig.configure(WebSecurityConfig.java:112) ~[traveliko-server-common-0.0.4-SNAPSHOT.jar:?]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.getHttp(WebSecurityConfigurerAdapter.java:199) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.init(WebSecurityConfigurerAdapter.java:290) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.init(WebSecurityConfigurerAdapter.java:69) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at com.traveliko.platform.server.common.config.WebSecurityConfig$$EnhancerBySpringCGLIB$$2fe26c57.init(<generated>) ~[traveliko-server-common-0.0.4-SNAPSHOT.jar:?]
	at org.springframework.security.config.annotation.AbstractConfiguredSecurityBuilder.init(AbstractConfiguredSecurityBuilder.java:371) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.AbstractConfiguredSecurityBuilder.doBuild(AbstractConfiguredSecurityBuilder.java:325) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.AbstractSecurityBuilder.build(AbstractSecurityBuilder.java:41) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration.springSecurityFilterChain(WebSecurityConfiguration.java:104) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10.CGLIB$springSecurityFilterChain$0(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10$$FastClassBySpringCGLIB$$a3f9a5eb.invoke(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228) ~[spring-core-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356) ~[spring-context-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10.springSecurityFilterChain(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 59 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher@725dfe92': Invocation of init method failed; nested exception is java.lang.UnsupportedOperationException: Section 4.4 of the Servlet 3.0 specification does not permit this method to be called from a ServletContextListener that was not defined in web.xml, a web-fragment.xml file nor annotated with @WebListener
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1578) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:399) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.security.config.annotation.configuration.AutowireBeanFactoryObjectPostProcessor.postProcess(AutowireBeanFactoryObjectPostProcessor.java:67) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry.createMvcMatchers(AbstractRequestMatcherRegistry.java:170) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.mvcMatchers(ExpressionUrlAuthorizationConfigurer.java:118) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.mvcMatchers(ExpressionUrlAuthorizationConfigurer.java:123) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at com.traveliko.platform.server.common.config.WebSecurityConfig.configure(WebSecurityConfig.java:112) ~[traveliko-server-common-0.0.4-SNAPSHOT.jar:?]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.getHttp(WebSecurityConfigurerAdapter.java:199) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.init(WebSecurityConfigurerAdapter.java:290) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.init(WebSecurityConfigurerAdapter.java:69) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at com.traveliko.platform.server.common.config.WebSecurityConfig$$EnhancerBySpringCGLIB$$2fe26c57.init(<generated>) ~[traveliko-server-common-0.0.4-SNAPSHOT.jar:?]
	at org.springframework.security.config.annotation.AbstractConfiguredSecurityBuilder.init(AbstractConfiguredSecurityBuilder.java:371) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.AbstractConfiguredSecurityBuilder.doBuild(AbstractConfiguredSecurityBuilder.java:325) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.AbstractSecurityBuilder.build(AbstractSecurityBuilder.java:41) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration.springSecurityFilterChain(WebSecurityConfiguration.java:104) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10.CGLIB$springSecurityFilterChain$0(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10$$FastClassBySpringCGLIB$$a3f9a5eb.invoke(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228) ~[spring-core-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356) ~[spring-context-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10.springSecurityFilterChain(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 59 more
Caused by: java.lang.UnsupportedOperationException: Section 4.4 of the Servlet 3.0 specification does not permit this method to be called from a ServletContextListener that was not defined in web.xml, a web-fragment.xml file nor annotated with @WebListener
	at org.apache.catalina.core.StandardContext$NoPluggabilityServletContext.getServletRegistrations(StandardContext.java:6592) ~[catalina.jar:8.5.4]
	at org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher.afterPropertiesSet(MvcRequestMatcher.java:128) ~[spring-security-web-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1637) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1574) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:399) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.security.config.annotation.configuration.AutowireBeanFactoryObjectPostProcessor.postProcess(AutowireBeanFactoryObjectPostProcessor.java:67) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry.createMvcMatchers(AbstractRequestMatcherRegistry.java:170) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.mvcMatchers(ExpressionUrlAuthorizationConfigurer.java:118) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.mvcMatchers(ExpressionUrlAuthorizationConfigurer.java:123) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at com.traveliko.platform.server.common.config.WebSecurityConfig.configure(WebSecurityConfig.java:112) ~[traveliko-server-common-0.0.4-SNAPSHOT.jar:?]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.getHttp(WebSecurityConfigurerAdapter.java:199) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.init(WebSecurityConfigurerAdapter.java:290) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter.init(WebSecurityConfigurerAdapter.java:69) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at com.traveliko.platform.server.common.config.WebSecurityConfig$$EnhancerBySpringCGLIB$$2fe26c57.init(<generated>) ~[traveliko-server-common-0.0.4-SNAPSHOT.jar:?]
	at org.springframework.security.config.annotation.AbstractConfiguredSecurityBuilder.init(AbstractConfiguredSecurityBuilder.java:371) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.AbstractConfiguredSecurityBuilder.doBuild(AbstractConfiguredSecurityBuilder.java:325) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.AbstractSecurityBuilder.build(AbstractSecurityBuilder.java:41) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration.springSecurityFilterChain(WebSecurityConfiguration.java:104) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10.CGLIB$springSecurityFilterChain$0(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10$$FastClassBySpringCGLIB$$a3f9a5eb.invoke(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228) ~[spring-core-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356) ~[spring-context-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$$EnhancerBySpringCGLIB$$555d0b10.springSecurityFilterChain(<generated>) ~[spring-security-config-4.1.2.RELEASE.jar:4.1.2.RELEASE]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.2.RELEASE.jar:4.3.2.RELEASE]
	... 59 more
```

**Comment by rwinch** [comment]
@bjornharvold Thanks for the report. I'm attempting to reproduce this, but am unable to at the moment. 

What application server / exact version are you using? Based on your stacktrace it appears to be Tomcat, but I cannot be certain and do not know which version.

**Comment by bjornharvold** [comment]
Hi @rwinch

I am using the 8.5.4 version of Tomcat. I went back to 4.1.1 and everything works fine again. Is there anything about my wildcard urls that could cause this w v4.1.2?

**Comment by rwinch** [comment]
@bjornharvold Thanks for the additional information. I was able to reproduce the issue and have provided a Pull Request to fix it. The Pull Request is ready for review by the team and will be merged into 4.1.x as soon as the review is complete

**Comment by bjornharvold** [comment]
Thank you for the fast response Rob!
